shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -787,61 +935,42 @@ static bool ClipRRectContainsPlatformViewBoundingRect(const SkRRect& clip_rrect,
   overlay_view_wrapper.accessibilityIdentifier =
       [NSString stringWithFormat:@"platform_view[%lld].overlay[%lld]", view_id, overlay_id];
 
-  UIView* overlay_view = layer->overlay_view.get();
+  UIView* overlay_view = this->overlay_view.get();
   // Set the size of the overlay view.
   // This size is equal to the device screen size.
-  overlay_view.frame = [flutter_view_.get() convertRect:flutter_view_.get().bounds
-                                                 toView:overlay_view_wrapper];
+  overlay_view.frame = [flutter_view convertRect:flutter_view.bounds toView:overlay_view_wrapper];
   // Set a unique view identifier, so the overlay_view can be identified in XCUITests.
   overlay_view.accessibilityIdentifier =
       [NSString stringWithFormat:@"platform_view[%lld].overlay_view[%lld]", view_id, overlay_id];
-
-  std::unique_ptr<SurfaceFrame> frame = layer->surface->AcquireFrame(frame_size_);
-  // If frame is null, AcquireFrame already printed out an error message.
-  if (!frame) {
-    return layer;
-  }
-  DlCanvas* overlay_canvas = frame->Canvas();
-  int restore_count = overlay_canvas->GetSaveCount();
-  overlay_canvas->Save();
-  overlay_canvas->ClipRect(rect);
-  overlay_canvas->Clear(DlColor::kTransparent());
-  slice->render_into(overlay_canvas);
-  overlay_canvas->RestoreToCount(restore_count);
-
-  // This flutter view is never the last in a frame, since we always submit the
-  // underlay view last.
-  frame->set_submit_info({.frame_boundary = false});
-
-  layer->did_submit_last_frame = frame->Submit();
-  return layer;
 }
 
-void FlutterPlatformViewsController::RemoveUnusedLayers() {
-  for (const auto& layer : layer_pool_->RemoveUnusedLayers()) {
+void FlutterPlatformViewsController::RemoveUnusedLayers(
+    const std::vector<std::shared_ptr<FlutterPlatformViewLayer>>& unused_layers,
+    const std::vector<int64_t>& composition_order,
+    const std::vector<int64_t>& active_composition_order) {
+  for (const std::shared_ptr<FlutterPlatformViewLayer>& layer : unused_layers) {
     [layer->overlay_view_wrapper removeFromSuperview];
   }
 
   std::unordered_set<int64_t> composition_order_set;
-  for (int64_t view_id : composition_order_) {
+  for (int64_t view_id : composition_order) {
     composition_order_set.insert(view_id);
   }
   // Remove unused platform views.
-  for (int64_t view_id : active_composition_order_) {
+  for (int64_t view_id : active_composition_order) {
     if (composition_order_set.find(view_id) == composition_order_set.end()) {
       UIView* platform_view_root = root_views_[view_id].get();
       [platform_view_root removeFromSuperview];
     }
   }
 }
 
-void FlutterPlatformViewsController::DisposeViews() {
+std::vector<UIView*> FlutterPlatformViewsController::DisposeViews() {
```

Comment:
DisposeViews is not inaccurately named because it doesn't actually dispose any views. How about `GetViewsToDispose`?

Reply:
Done